src/Tasks/Microsoft.NET.Build.Tasks/ResolveReadyToRunCompilers.cs

```diff
@@ -203,6 +203,10 @@ private bool GetCrossgen2TargetOS(out string targetOS)
                 {
                     portablePlatform = "osx";
                 }
+                else if (RuntimeInformation.IsOSPlatform(OSPlatform.FreeBSD))
```

Comment:
This whole section is very confusing. It maps Windows to Linux, and Linux to Windows. It suggests that it is not really used. Is it really necessary to add FreeBSD here?

I think it would be best to handle `portablePlatform == null` as generic Unix.

Reply:
It was added https://github.com/dotnet/sdk/commit/71e047e2adbbd8a714e4a3df9c82f08b86ba148c as a part of source-build so I assumed that it was used and the removal in the prior attempt to add in FreeBSD was the cause of the source-build (centos) issue. 

`OSPlatform` does contain `FreeBSD` but only for [netcore 3+](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.osplatform.freebsd?view=net-7.0#system-runtime-interopservices-osplatform-freebsd). This looks like it causes the build to run into the same issue when trying to add FreeBSD in other areas (like runtime).

Not sure how to approach the rest of this block. Would it be best to just have `portablePlatform = "unix";` for `portablePlatform == null && _targetRuntimeIdentifier == _hostRuntimeIdentifier` instead of the `if` statements? A different case for just `portablePlatform == null` -> `portablePlatform = "unix";` and leave the current `if` alone?

Reply:
Is the PR in the current state fixing the source build problem for FreeBSD? If it does, it looks good to me.

I think there is opportunity to make this work for all Unix-like OSes out of the box, but it does not look easy. We can leave it for later.

Reply:
Yes. As-is the issues are fixed for FreeBSD